### PR TITLE
device feature: notify only specific devices

### PR DESCRIPTION
**Pull request by r41d**
For now, support has only been added for pushover.

### CODE REVIEW

**Comment by r41d** [comment]
Not related to this PR but something I noticed: The first line of the main script should probably say
```
#!/usr/bin/env python2
```
with a 2 at the end, because the code is python2 only.

**Comment by jpmens** [comment]
Thanks for your idea / contribution. Why does this addition require modifying mqttwarn.py and not simply use the pushover target configuration? In particular because it is used by pushover only.

**Comment by r41d** [comment]
Good question. My first approach was to incorporate this into Pushover by changing the data structure from a list to a dict. The dict could then include 'sound' and 'device' if desired. However, this would break all existing configurations using Pushover, which is bad, so I started over.
Just like specifying a priority, I would consider it a generic feature, e.g. Pushbullet supports it too. I just didn't have experience with any other service, that's why I only included support for Pushover at this time.

**Comment by jpmens** [comment]
Without having thought about this very much, I think this is too invasive as it stands, and I don't understand why you don't add this _device_ as an extra addr. Looking at the code of the current `pushover.py` service

```python
    # addrs is an array with two or three elements:
    # 0 is the user key
    # 1 is the app key
    # 2 (if present) is the PushOver sound to play for the message

    try:
        userkey = addrs[0]
        appkey  = addrs[1]
    except:
        srv.logging.warn("No pushover userkey/appkey configured for target `%s'" % (item.target))
        return False

...
    if len(addrs) > 2:
        params['sound'] = addrs[2]
```

it seems to me you could add it `if len(addrs) > 3` or similar.

**Comment by jpmens** [comment]
Let people specify sound as `nil` or something, check and ignore that value.

**Comment by r41d** [comment]
OK, I changed it, looking forward for review

File: README.md [hunk 1]
```diff
@@ -1994,6 +1995,9 @@ notify, say, one or more of your devices as well as one for your spouse. As you
 can see in the example, you can even specify an optional sound to be played for
 the individual users. For a list of available sounds see the [Pushover API List](https://pushover.net/api#sounds).
 
+You can also specify the devices that should be notified, this is a comma-separated list.
```
Comment (jpmens) [inline README.md]: please add "comma-separated list of device names specified as a single string" to ensure people don't try to add a Python-like array or something.

File: mqttwarn.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
```
Comment (jpmens) [inline mqttwarn.py]: Kindly undo that change for the time being. That path may be correct for your system but it isn't in general.

File: README.md (hunk 1 above)
Comment (r41d) [reply README.md]: Sure, more clarification is always good

File: mqttwarn.py (hunk 2 above)
Comment (r41d) [reply mqttwarn.py]: Thought this was pretty general for Linux, but sure, will change it

Comment (jpmens) [reply mqttwarn.py]: It may well be, but there's not just Linux in the world. :-)

**Comment by jpmens** [comment]
Merged, thank you!

**Pull request merged**